### PR TITLE
fix tests

### LINKED ISSUE #3360: Fix failing tests

### PR DESCRIPTION
**Pull request by Fraser999**
Fixes various failing unit tests, removes the remaining `allow(unused)` instances and removes corresponding unused code.

Closes #3360.

### CODE REVIEW

File: node/src/reactor/main_reactor/tests.rs [hunk 1]
```diff
@@ -107,7 +102,7 @@ impl TestChain {
         chainspec.network_config.accounts_config = AccountsConfig::new(accounts, delegators);
 
         // Make the genesis timestamp 60 seconds from now, to allow for all validators to start up.
```
Comment (afck) [inline node/src/reactor/main_reactor/tests.rs:104]:
```suggestion
        // Make the genesis timestamp 10 seconds from now, to allow for all validators to start up.
```

File: node/src/types/block.rs [hunk 2]
```diff
@@ -913,6 +913,35 @@ impl BlockHeader {
...
+            state_root_hash,
+            body_hash,
+            random_bit,
+            accumulated_seed,
+            era_end,
+            timestamp,
+            era_id,
+            height,
+            protocol_version,
+            block_hash: _,
```
Comment (afck) [inline node/src/types/block.rs:930]: (Right, I missed that! Good catch!)

**Review by afck — APPROVED** [review]
Looks great! :+1:

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: node/src/components/event_stream_server.rs [hunk 3]
```diff
@@ -234,64 +295,7 @@ where
         &mut self,
         _effect_builder: EffectBuilder<REv>,
     ) -> Result<Effects<Self::ComponentEvent>, Self::Error> {
-        let cfg = self.config.clone();
```
Comment (EdHastingsCasperAssociation) [inline node/src/components/event_stream_server.rs:237]: just cuz?

**Review by EdHastingsCasperAssociation — APPROVED** [review]

File: node/src/reactor/main_reactor/tests.rs (hunk 1 above)
Comment (rafal-ch) [reply node/src/reactor/main_reactor/tests.rs:104]: Fixed in https://github.com/casper-network/casper-node/pull/3396/commits/12a53e511c8a0dc8befb3538bb1a1243b25b5717

File: node/src/components/event_stream_server.rs (hunk 3 above)
Comment (rafal-ch) [reply node/src/components/event_stream_server.rs:237]: I moved the call to `clone()` to the last consumer in https://github.com/casper-network/casper-node/pull/3396/commits/12a53e511c8a0dc8befb3538bb1a1243b25b5717

**Pull request merged**